File: backend/lib/L1_domain/entities/goals/task.py
#  Copyright (c) 2022. Alexandr Moroz

from __future__ import annotations

from dataclasses import dataclass

from .smartable import Smartable
from .task_priority import TaskPriority
from .task_status import TaskStatus


@dataclass
class Task(Smartable):
    status: TaskStatus | None = None
    priority: TaskPriority | None = None
    tasks: list[Task] | None = None

File: frontend/lib/L2_data/mappers/task.dart
// Copyright (c) 2022. Alexandr Moroz

import 'package:openapi/openapi.dart';

import '../../L1_domain/entities/goals/task.dart';
import 'task_status.dart';

extension TaskMapper on TaskSchemaGet {
  Task get task => Task(
        id: id,
        parentId: parentId,
        createdOn: createdOn.toLocal(),
        updatedOn: updatedOn.toLocal(),
        title: title,
        description: description ?? '',
        dueDate: dueDate?.toLocal(),
        status: status != null ? status!.status : null,
        tasks: tasks?.map((t) => t.task) ?? [],
      );
}
